### ISSUE TITLE
upspin-ui: feature request: make vm name configurable during setup

### ISSUE THREAD

**Issue by amygdala** [upspin/upspin#548]
In the browser-based setup flow, it would be great to be able to define the name of the vm (rather than having it fixed to 'upspin'.  
This would have allowed me to retain my existing server for later diagnosis, during the process of setting up a new one, as AFAICT GCP does not allow vm instance rename.

**Comment by adg** [comment]
This would be useful, although I want to avoid overwhelming the user with options during the setup process. There are some other GCP resources with default names, too:

- an allocated IP address named `upspinserver`
- a persistent disk named `upspinserver`
- a firewall rule named `allow-https`

The first two could just share the name of the VM instance (as they do now). I'm not sure if it's worth making it possible to specify an alternate name for the firewall rule; if one exists by that name already it likely does what we want it to do anyway.

But I'm still on the fence about making it configurable at all. The rationale behind making these hard-coded values was the assumption that the user will create a dedicated cloud project for their upspinserver deployment. In your case, you could have kept the old instance running by creating a new cloud project for your second deployment. What do you think?

(This is related to #517, in that it is also about specifying an additional option when configuring the VM.)

**Comment by amygdala** [comment]
Hmm. I think my particular case was kind of a one-off.  (Most people won't manage to trigger weird errors so soon after deploying and want to just start over :). So in that sense I agree it's not worth thinking about as a high priority.

(Though -- I'm not sure that your assumption of one deploy per project will hold in the longer term. It will be interesting to see how/if people build more managed services on top of upspin).

**Comment by adg** [comment]
Agreed. I'll leave this issue open as a reminder to revisit one day.
